### PR TITLE
Fix root-relative xpath expressions on cloned nodes

### LINKED ISSUE #62: XPath "//" doesn't work on cloned nodes
This test doesn't work:

```java
final XML xml = new XMLDocument("<t6><z9 a='433'/></t6>");
final XML root = xml.nodes("/t6").get(0);
MatcherAssert.assertThat(
  root.xpath("//z9/@a").get(0),
  Matchers.equalTo("433")
);
```

### PR DESCRIPTION
**Pull request by Suseika**
Fixes #62

### CODE REVIEW

**Comment by dmarkov** [comment]
@suseika Thanks, let me find someone who can review this pull request

**Comment by dmarkov** [comment]
@krzyk please review this one

File: src/main/java/com/jcabi/xml/XMLDocument.java [hunk 1]
```diff
@@ -155,7 +155,7 @@
      * @param text XML document body
      */
     public XMLDocument(@NotNull(message = "XML text can't be NULL")
-        final String text) {
+                       final String text) {
```
Comment (krzyk) [inline src/main/java/com/jcabi/xml/XMLDocument.java]: @Suseika we don't indent arguments that far, please make it just 4 spaces as it was before, here and in the rest of the places you have changed

File: src/main/java/com/jcabi/xml/XMLDocument.java [hunk 2]
```diff
@@ -293,7 +294,15 @@ public String toString() {
     @Override
     @NotNull(message = "node is never NULL")
     public Node node() {
-        return Node.class.cast(this.cache).cloneNode(true);
+        final Node castCache = Node.class.cast(this.cache);
+        final Node answer;
+        final boolean isDocument = castCache instanceof Document;
```
Comment (krzyk) [inline src/main/java/com/jcabi/xml/XMLDocument.java]: @Suseika please inline `isDocument`

File: src/main/java/com/jcabi/xml/XMLDocument.java [hunk 3]
```diff
@@ -377,11 +386,30 @@ public XML registerNs(@NotNull final String prefix,
...
     }
 
     /**
+     * Clones a node importing the clone in a new document.
+     *
+     * @param node A node to clone.
+     * @return A cloned node imported in a dedicated document.
+     */
+    private Node createImportedNode(final Node node) {
+        assert !(node instanceof Document);
```
Comment (krzyk) [inline src/main/java/com/jcabi/xml/XMLDocument.java]: @Suseika please remove `assert`, we don't use them

**Comment by krzyk** [comment]
@Suseika please see my comments above

**Comment by Suseika** [comment]
@krzyk Done. Is it mergeable now?

File: src/main/java/com/jcabi/xml/XMLDocument.java [hunk 4]
```diff
@@ -382,6 +389,24 @@ public XML merge(@NotNull(message = "context can't be NULL")
     }
 
     /**
+     * Clones a node importing the clone in a new document.
```
Comment (krzyk) [inline src/main/java/com/jcabi/xml/XMLDocument.java]: @Suseika maybe *Clones a node and imports it into a new document*

**Comment by krzyk** [comment]
@Suseika just slight wording change

**Comment by Suseika** [comment]
@krzyk Done.

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by yegor256** [comment]
@rultor merge

**Comment by dmarkov** [comment]
@krzyk thanks, I just added **18 mins** to your account, payment `59134643`, 25 hours and 24 mins spent

extra minutes for review comments (c=3)

+18 added to your rating, current score is: [+4597](http://www.netbout.com/b/35035?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy now pls

**Pull request merged**